ipamd/ipamd.go

```diff
@@ -400,23 +418,44 @@ func (c *IPAMContext) waitENIAttached(eni string) (awsutils.ENIMetadata, error)
 	}
 }
 
+func getWarmENITarget() int {
+	inputStr, found := os.LookupEnv("WARM_ENI_TARGET")
+
+	if !found {
+		return defaultWarmENITarget
+	}
+
+	if input, err := strconv.Atoi(inputStr); err == nil {
+		if input < 0 {
+			return defaultWarmENITarget
+		}
+		log.Debugf("Using WARM-ENI-TARGET %v", input)
+		return input
+	}
+	return defaultWarmENITarget
+}
+
+func logPoolStats(total, used, currentMaxAddrsPerENI, maxAddrsPerENI int) {
+	log.Debugf("IP pool stats: total=%d, used=%d, c.currentMaxAddrsPerENI =%d, c.maxAddrsPerENI = %d",
+		total, used, currentMaxAddrsPerENI, maxAddrsPerENI)
+}
+
 //nodeIPPoolTooLow returns true if IP pool is below low threshhold
 func (c *IPAMContext) nodeIPPoolTooLow() bool {
+	warmENITarget := getWarmENITarget()
 	total, used := c.dataStore.GetStats()
-	log.Debugf("IP pool stats: total=%d, used=%d, c.currentMaxAddrsPerENI =%d, c.maxAddrsPerENI = %d",
-		total, used, c.currentMaxAddrsPerENI, c.maxAddrsPerENI)
+	logPoolStats(total, used, c.currentMaxAddrsPerENI, c.maxAddrsPerENI)
 
-	return ((total - used) <= c.currentMaxAddrsPerENI)
+	return ((total - used) <= c.currentMaxAddrsPerENI*warmENITarget)
```

Comment:
this would be clearer if you said `available := total - used` then made the comparison against available.  add space around ' * '.

Reply:
go fmt removes spaces around '*'